kani-compiler/src/kani_middle/transform/check_uninit/ty_layout.rs

```diff
@@ -332,3 +332,36 @@ fn data_bytes_for_ty(
         FieldsShape::Array { .. } => Ok(vec![]),
     }
 }
+
+/// Returns true if `to_ty` has a smaller or equal size and the same padding bytes as `from_ty` up until
+/// its size.
+pub fn tys_layout_compatible(from_ty: &Ty, to_ty: &Ty) -> bool {
```

Comment:
This name is not clear to me. Is this trying to compute where it is safe to transmute `from_ty` to `to_ty`? Or is this trying to compute where it is safe to transmute in both directions?

Reply:
I fixed that by adding a distinction in the commit handling `transmute`